### DISCUSSION TITLE
Revalidate page cache on user actions

### DISCUSSION THREAD

**Discussion by uchar** [vercel/next.js#15998] [Help]
I'm currently using SSG with the new next.js `revalidate` feature for making a question/answer forum 

It works fine but the problem is when the user answers a new question and refreshes the page, his answer doesn't load until his specified time expires. so I need a way to revalidate page on some user actions. 

Is it possible?

**Comment by matamatanot** [comment]
@uchar 
Not yet. But,
https://github.com/vercel/next.js/discussions/11552#discussioncomment-2658

> Revalidation on an interval is just the beginning for this feature! Using the same keys mentioned above, we could allow you to trigger revalidations on-demand using an API endpoint.
> This would change the model to be event-driven instead of interval-based.

**Reply by matamatanot** [reply]
https://github.com/vercel/next.js/discussions/11552#discussioncomment-2786
